src/Features/Core/Portable/Microsoft.CodeAnalysis.Features.csproj

```diff
@@ -50,6 +50,7 @@
     <InternalsVisibleTo Include="Microsoft.CodeAnalysis.ExternalAccess.Razor" />
     <InternalsVisibleTo Include="Microsoft.CodeAnalysis.ExternalAccess.Razor.UnitTests" />
     <InternalsVisibleTo Include="Microsoft.CodeAnalysis.ExternalAccess.UnitTesting" />
+    <InternalsVisibleTo Include="Microsoft.CodeAnalysis.ExternalAccess.VisualDiagnostics" />
```

Comment:
if this is just temp its fine, but def everything you need should be exposed via the externalaccess.xaml proj

Reply:
I would like to double check something. EA.XAML is supposed to be XAML intellisense only. EA.VisualDignostics is supposed to be XAML Hot Reload and few other hot reload flavors. Thus EA.VD should be able accessing IDiagnosticSource API. But I can definitely route everything via EA.XAML

Reply:
my bad, forgot that ea.visualsdiagnostics was the ea for xaml hot reload.  this is fine